tests/unittests/OperatorTest.cpp

```diff
@@ -6172,6 +6173,72 @@ TEST_P(OperatorTest, SigmoidSweep) {
   testSigmoidFp16Sweep(bindings_, mod_, F_, EE_);
 }
 
+/// Reference ideal tanh implementation. Computes an fp32 tanh
+/// and casts the result to FP16, no denorms
+static float16_t refTanHFp16(float x) {
+  float res = (exp(2 * x) - 1) / (exp(2 * x) + 1);
+  if (fabs(res) < 6e-5) {
+    res = 0.0;
+  }
+  return (float16_t)res;
+}
+
+/// Test to verify that the tanh implementation is close to the ideal one
+/// Does a sweep of -15,15 and prints the outputs of the NNPI implementation
+/// compared to the ideal tanh in fp16.
+static void testTanHFp16Sweep(glow::PlaceholderBindings &bindings,
+                              glow::Module &mod, glow::Function *F,
+                              glow::ExecutionEngine &EE) {
+  constexpr size_t N = 100;
+  auto *input = mod.createPlaceholder(ElemKind::FloatTy, {N}, "input", false);
+  auto inputH = bindings.allocate(input)->getHandle();
+
+  constexpr float range_start = -15;
+  constexpr float range_end = 15;
+  constexpr float delta = (range_end - range_start) / N;
+
+  for (size_t i = 0; i < N; i++) {
+    inputH.raw(i) = range_start + i * delta;
+  }
+
+  auto *sigmoid = F->createTanh("TanH", input);
+  auto *save = F->createSave("Save", sigmoid);
+  auto *resultTensor = bindings.allocate(save->getPlaceholder());
+
+  CompilationContext cctx;
+  cctx.precisionConfig.convertToFP16 = true;
+  cctx.precisionConfig.convertFusedToFP16 = true;
+
+  EE.compile(cctx);
+  EE.run(bindings);
+
+  auto resultH = resultTensor->getHandle();
+  int count = 0;
+
+  for (size_t i = 0; i < N; i++) {
+    float input_v = inputH.at({i});
+    float ref_ideal = refTanHFp16(input_v);
+    float output = resultH.at({i});
+    float diff = fabs(output - ref_ideal);
+
+    if (diff > 1e-6) {
+      count++;
+    }
+
+    llvm::outs() << "TanH " << i << " " << input_v << " Backend:" << output
+                 << " ref_ideal:" << ref_ideal << " diff:" << diff << "\n";
+  }
+  llvm::outs().flush();
+
+  EXPECT_EQ(count, 0);
+}
+
+TEST_P(OperatorTest, TanHSweep) {
```

Comment:
nit: Can we name this `TanHSweep_Float16` or something, so it's clear from the name the test is specific for FP16? Note this will require changing the blacklists. Maybe could do for your sigmoid test too.